src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -49,11 +49,13 @@ public static IntPtr OffsetOf(Type t, string fieldName)
                 throw new ArgumentException(SR.Argument_MustBeRuntimeFieldInfo, nameof(fieldName));
             }
 
-            return OffsetOfHelper(rtField);
+            nint offset = OffsetOfInternal(rtField.GetFieldHandle());
+            GC.KeepAlive(rtField);
+            return offset;
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern IntPtr OffsetOfHelper(IRuntimeFieldInfo f);
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "MarshalNative_OffsetOfInternal")]
+        private static partial nint OffsetOfInternal(IntPtr pFD);
```

Comment:
```suggestion
        private static partial nint OffsetOf(IntPtr pFD);
```
Nit: The internal suffix is not necessary. The signature is sufficient to make the internal entrypoint distinct.